src/main/java/net/pcal/mobfilter/MFService.java

```diff
@@ -79,6 +77,28 @@ public boolean isSpawnAllowed(ServerWorld sw,
         }
         return allowSpawn;
     }
+    public boolean isSpawnAllowed(ServerWorld sw,
+                                  SpawnGroup sg,
+                                  SpawnSettings.SpawnEntry se,
+                                  BlockPos.Mutable pos) {
+        return isSpawnAllowed(sw, sg, se.type, pos);
+    }
+    public boolean isSpawnAllowed(WorldView world, BlockPos pos, EntityType<?> entityType) {
+        ServerWorld serverWorld;
+        // I went through and looked at all the usages of the mixin that calls this and the methods that call this always use ServerWorldAccess or ServerWorld
+        if (world instanceof ServerWorldAccess) {
+            serverWorld = ((ServerWorldAccess) world).toServerWorld();
+        } else if (world instanceof  ServerWorld) {
+            serverWorld = ((ServerWorld) world);
+        } else {
+            // We shouldn't ever get here but I put it here just in case
+            this.logger.warn("Failed to check the rules for an {} entity. Couldn't find a valid cast for {}", entityType.getName().getString(), world.getClass().getName());
+            return true;
+        }
+        SpawnGroup spawnGroup = entityType.getSpawnGroup();
+
+        return isSpawnAllowed(serverWorld, spawnGroup, entityType, pos.mutableCopy());
```

Comment:
I’m not quite sure but if it works without the copy then great. I think I might have done it because the last argument is a BlockPos.Mutable and I wasn’t quite sure why.